rs/replicated_state/src/canister_snapshots.rs

```diff
@@ -27,10 +27,14 @@ pub struct CanisterSnapshots {
 }
 
 impl CanisterSnapshots {
-    pub fn new(
-        snapshots: BTreeMap<SnapshotId, Arc<CanisterSnapshot>>,
-        snapshot_ids: BTreeMap<CanisterId, BTreeSet<SnapshotId>>,
-    ) -> Self {
+    pub fn new(snapshots: BTreeMap<SnapshotId, Arc<CanisterSnapshot>>) -> Self {
+        let mut snapshot_ids = BTreeMap::default();
+        for snapshot_id in snapshots.keys() {
+            let canister_id = snapshot_id.get_canister_id();
+            let canister_snapshot_ids: &mut BTreeSet<SnapshotId> =
```

Comment:
Using a `BTreeSet` here will implicitly discard duplicates. Is this what we want here? If so should we maybe add a comment that makes it clear why?